src/main/java/com/aws/greengrass/device/session/SessionManager.java

```diff
@@ -23,13 +24,22 @@ public class SessionManager {
     private static final Logger logger = LogManager.getLogger(SessionManager.class);
     private static final String SESSION_ID = "SessionId";
 
-    // TODO: Add mechanism to prevent session leaks
-    // This implementation currently relies on clients to properly close sessions. If they don't, then sessions
-    // will be leaked. We can add this once sessions are created with the entire set of device credentials. However,
+    // LRU Session Cache that evicts the eldest entry (based on access order) upon reaching its size.
+    // TODO: Support time-based cache eviction (Session timeout) and Session deduping.
+    // We can add this once sessions are created with the entire set of device credentials. However,
     // Moquette is currently creating sessions using just the certificate. Since multiple clients could use the same
     // certificate, we can't de-dup based on this alone.
     @Getter(AccessLevel.PACKAGE)
-    private final Map<String, Session> sessionMap = new ConcurrentHashMap<>();
+    private final Map<String, Session> sessionMap = Collections.synchronizedMap(
+            new LinkedHashMap<String, Session>(getSessionCapacity(), 0.75f, true) {
+                @Override
+                protected boolean removeEldestEntry(Map.Entry<String, Session> eldest) {
```

Comment:
Sessions will be getting automatically removed. Ensure we don't need any cleanup logic when sessions are dropped.

Reply:
Good idea, yes. Maybe trace, just so we have it if absolutely needed